From 18f295f9ad9edcf3858ed6efab07a9fa02f8c0e0 Mon Sep 17 00:00:00 2001
From: Mohamed Sentissi <msentissi@anaconda.com>
Date: Thu, 24 Oct 2024 13:40:23 -0400
Subject: [PATCH] skip failures

We patch out some tests because they expect pandoc 3.2 as a dependency (the version on CF), but we have an earlier version on defaults.

One failed because it needs a PDF engine such as pdflatex, which is optional.

---
 mkdocs/api.md           |  24 ---------
 mkdocs/configuration.md |   8 ---
 mkdocs/cookbook.md      |  31 +----------
 mkdocs/document.md      |  18 -------
 mkdocs/examples.md      |  26 ----------
 mkdocs/iteration.md     |   9 ----
 mkdocs/markdown.md      | 112 ----------------------------------------
 7 files changed, 1 insertion(+), 227 deletions(-)

diff --git a/mkdocs/api.md b/mkdocs/api.md
index 0157ced..0b274e6 100644
--- a/mkdocs/api.md
+++ b/mkdocs/api.md
@@ -182,9 +182,6 @@ from pandoc.types import *
     >>> _ = pandoc.write(doc, file="doc.html")
     >>> open("doc.html", encoding="utf-8").read()
     '<p>Hello world!</p>\n'
-    >>> _ = pandoc.write(doc, file="doc.pdf")
-    >>> open("doc.pdf", "rb").read() # doctest: +ELLIPSIS
-    b'%PDF...'
     ```
 
     Use extra pandoc options:
@@ -346,30 +343,9 @@ from pandoc.types import *
     call `pandoc.read` or `pandoc.write` and will automatically infer the
     configuration from the `pandoc` executable found in the path (or fails).
 
-    ``` pycon
-    >>> config = pandoc.configure(read=True)
-    >>> config # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
-    {'auto': True, 
-     'path': ..., 
-     'version': '3.2.1', 
-     'pandoc_types_version': '1.23.1'}
-    ```
     To avoid this, call `pandoc.configure(...)` yourself beforehand.
     Alternatively, select manually your pandoc executable afterwards:
 
-    ``` pycon
-    >>> pandoc.configure(reset=True)
-    >>> pandoc.configure(read=True) is None
-    True
-    >>> config["auto"] = False
-    >>> pandoc.configure(**config)
-    >>> pandoc.configure(read=True) # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
-    {'auto': False, 
-     'path': ..., 
-     'version': '3.2.1', 
-     'pandoc_types_version': '1.23.1'}    
-    ```
-
     <h5>See also</h5>
 
     Refer to the [Configuration section](../configuration).
diff --git a/mkdocs/configuration.md b/mkdocs/configuration.md
index f53f198..f9be196 100644
--- a/mkdocs/configuration.md
+++ b/mkdocs/configuration.md
@@ -78,14 +78,6 @@ does not change the current configuration
 but returns a dictionary whose keys are `auto`, `path`, 
 `version` and `pandoc_types_version`, such as
 
-``` pycon
->>> pandoc.configure(read=True) # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
-{'auto': True, 
- 'path': ..., 
- 'version': '3.2.1', 
- 'pandoc_types_version': '1.23.1'}
-```
-
 The `read` option may be combined with other arguments, for example
 
     config = pandoc.configure(auto=True, read=True)
diff --git a/mkdocs/cookbook.md b/mkdocs/cookbook.md
index 80ff4af..b1b5272 100644
--- a/mkdocs/cookbook.md
+++ b/mkdocs/cookbook.md
@@ -217,16 +217,6 @@ it yields
 
 ### Pattern matching
 
-With Python 3.10 or later, [pattern matching] can be used to combine
-random access and structural checks. The following implementation of `get_date`
-
-``` python
-def get_date(doc):
-    match doc:
-        case Pandoc(Meta({"date": MetaInlines(date_inlines)}), _):
-            return pandoc.write(date_inlines).strip()
-```
-
 and the previous one have identical behaviors:
 
 ``` pycon
@@ -237,13 +227,6 @@ and the previous one have identical behaviors:
 
 The behavior of the following `get_first_header_title` function
 
-``` python
-def get_first_header_title(doc):
-    match doc:
-        case Pandoc(_, [Header(_, _, header_inlines), *_]):
-            return pandoc.write(header_inlines).strip()
-```
-
 is also unchanged:
 
 ``` pycon
@@ -1005,19 +988,7 @@ Inline = ...
        | Span(Attr, [Inline])
 ```
 
-and 
-
-``` pycon
->>> Block  # doctest: +ELLIPSIS
-Block = ...
-      | CodeBlock(Attr, Text)
-      ...
-      | Header(Int, Attr, [Inline])
-      ...
-      | Table(Attr, Caption, [ColSpec], TableHead, [TableBody], TableFoot)
-      ...
-      | Div(Attr, [Block])
-```
+and
 
 So we need to target `Code`, `Link`, `Image`, `Span`, `Div`,`CodeBlock`,
 `Header`, `Table` and `Div` instances. `Header` is a special case here 
diff --git a/mkdocs/document.md b/mkdocs/document.md
index 07c70ea..c813009 100644
--- a/mkdocs/document.md
+++ b/mkdocs/document.md
@@ -149,24 +149,6 @@ to conclude that the fragment is valid.
 Now, let's explore the content of the document which is defined as a list of
 blocks. The `Block` type signature is
 
-``` pycon
->>> Block
-Block = Plain([Inline])
-      | Para([Inline])
-      | LineBlock([[Inline]])
-      | CodeBlock(Attr, Text)
-      | RawBlock(Format, Text)
-      | BlockQuote([Block])
-      | OrderedList(ListAttributes, [[Block]])
-      | BulletList([[Block]])
-      | DefinitionList([([Inline], [[Block]])])
-      | Header(Int, Attr, [Inline])
-      | HorizontalRule()
-      | Table(Attr, Caption, [ColSpec], TableHead, [TableBody], TableFoot)
-      | Figure(Attr, Caption, [Block])
-      | Div(Attr, [Block])
-```
-
 Each `"|"` symbol in the signature represents an alternative: blocks are 
 either instances of `Plain` or `Para` or `LineBlock`, etc. In our example
 document, the only type of block that was used is the paragraph type `Para`,
diff --git a/mkdocs/examples.md b/mkdocs/examples.md
index f3aabaa..4c1c32c 100644
--- a/mkdocs/examples.md
+++ b/mkdocs/examples.md
@@ -148,17 +148,6 @@ def is_theorem(elt):
     return False
 ```
 
-Or equivalenty, with Python 3.10 (or newer), using pattern matching:
-
-``` python
-def is_theorem(elt):
-    match elt:
-        case Div((_, classes, _), _) if "theorem" in classes:
-            return True
-        case _:
-            return False
-```
-
 Now we can implement the transformation itself:
 
 ``` python
@@ -188,21 +177,6 @@ $$f(z) = \frac{1}{i2\pi} \int \frac{f(w)}{w-z}\, dw$$
 """
 ```
 
-``` pycon
->>> doc = pandoc.read(markdown)
->>> print(pandoc.write(doc, format="latex")) # doctest: +NORMALIZE_WHITESPACE
-\phantomsection\label{cauchy-formula}
-\[f(z) = \frac{1}{i2\pi} \int \frac{f(w)}{w-z}\, dw\]
->>> theoremize(doc)
->>> print(pandoc.write(doc, format="latex")) # doctest: +NORMALIZE_WHITESPACE
-\phantomsection\label{cauchy-formula}
-\begin{theorem}\label{cauchy-formula}
-<BLANKLINE>
-\[f(z) = \frac{1}{i2\pi} \int \frac{f(w)}{w-z}\, dw\]
-<BLANKLINE>
-\end{theorem}
-```
-
 Jupyter Notebooks
 --------------------------------------------------------------------------------
 
diff --git a/mkdocs/iteration.md b/mkdocs/iteration.md
index d0841e5..6bc17bb 100644
--- a/mkdocs/iteration.md
+++ b/mkdocs/iteration.md
@@ -127,15 +127,6 @@ world!
 With Python 3.10 (or newer), pattern matching can be used for every
 Pandoc element:
 
-``` pycon
->>> doc = pandoc.read("Hello world!")
->>> match doc:
-...     case Pandoc(Meta(meta), [Para(inlines)]):
-...         assert meta == {}
-...         print(inlines)
-[Str('Hello'), Space(), Str('world!')]
-```
-
 Tree Iteration
 --------------------------------------------------------------------------------
 
diff --git a/mkdocs/markdown.md b/mkdocs/markdown.md
index 94cace2..af08b46 100644
--- a/mkdocs/markdown.md
+++ b/mkdocs/markdown.md
@@ -537,34 +537,6 @@ This is equivalent to:
 
 <!-- prevent container tabs merge -->
 
-This shortcut form may be combined with attributes:
-
-=== "Markdown"
-
-        ```haskell {.numberLines}
-        qsort [] = []
-        ```
-
-=== "Python"
-
-        Pandoc(Meta({}), [CodeBlock(('', ['haskell', 'numberLines'], []), 'qsort [] = []')])
-
-<!-- prevent container tabs merge -->
-
-Which is equivalent to:
-
-=== "Markdown"
-
-        ``` {.haskell .numberLines}
-        qsort [] = []
-        ```
-
-=== "Python"
-
-        Pandoc(Meta({}), [CodeBlock(('', ['haskell', 'numberLines'], []), 'qsort [] = []')])
-
-<!-- prevent container tabs merge -->
-
 If the `fenced_code_attributes` extension is disabled, but input
 contains class attribute(s) for the code block, the first class
 attribute will be printed after the opening fence as a bare word.
@@ -1277,52 +1249,6 @@ Grid tables look like this:
 
 <!-- prevent container tabs merge -->
 
-The row of `=`s separates the header from the table body, and can be
-omitted for a headerless table. The cells of grid tables may contain
-arbitrary block elements (multiple paragraphs, code blocks, lists,
-etc.).
-
-Cells can span multiple columns or rows:
-
-=== "Markdown"
-
-        +---------------------+----------+
-        | Property            | Earth    |
-        +=============+=======+==========+
-        |             | min   | -89.2 °C |
-        | Temperature +-------+----------+
-        | 1961-1990   | mean  | 14 °C    |
-        |             +-------+----------+
-        |             | min   | 56.7 °C  |
-        +-------------+-------+----------+
-
-=== "Python"
-
-        Pandoc(Meta({}), [Table(('', [], []), Caption(None, []), [(AlignDefault(), ColWidth_(0.19444444444444445)), (AlignDefault(), ColWidth_(0.1111111111111111)), (AlignDefault(), ColWidth_(0.1527777777777778))], TableHead(('', [], []), [Row(('', [], []), [Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(2), [Plain([Str('Property')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('Earth')])])])]), [TableBody(('', [], []), RowHeadColumns(0), [], [Row(('', [], []), [Cell(('', [], []), AlignDefault(), RowSpan(3), ColSpan(1), [Plain([Str('Temperature'), SoftBreak(), Str('1961-1990')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('min')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('-89.2'), Space(), Str('°C')])])]), Row(('', [], []), [Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('mean')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('14'), Space(), Str('°C')])])]), Row(('', [], []), [Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('min')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('56.7'), Space(), Str('°C')])])])])], TableFoot(('', [], []), []))])
-
-<!-- prevent container tabs merge -->
-
-A table header may contain more than one row:
-
-=== "Markdown"
-
-        +---------------------+-----------------------+
-        | Location            | Temperature 1961-1990 |
-        |                     | in degree Celsius     |
-        |                     +-------+-------+-------+
-        |                     | min   | mean  | max   |
-        +=====================+=======+=======+=======+
-        | Antarctica          | -89.2 | N/A   | 19.8  |
-        +---------------------+-------+-------+-------+
-        | Earth               | -89.2 | 14    | 56.7  |
-        +---------------------+-------+-------+-------+
-
-=== "Python"
-
-        Pandoc(Meta({}), [Table(('', [], []), Caption(None, []), [(AlignDefault(), ColWidth_(0.3055555555555556)), (AlignDefault(), ColWidth_(0.1111111111111111)), (AlignDefault(), ColWidth_(0.1111111111111111)), (AlignDefault(), ColWidth_(0.1111111111111111))], TableHead(('', [], []), [Row(('', [], []), [Cell(('', [], []), AlignDefault(), RowSpan(2), ColSpan(1), [Plain([Str('Location')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(3), [Plain([Str('Temperature'), Space(), Str('1961-1990'), SoftBreak(), Str('in'), Space(), Str('degree'), Space(), Str('Celsius')])])]), Row(('', [], []), [Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('min')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('mean')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('max')])])])]), [TableBody(('', [], []), RowHeadColumns(0), [], [Row(('', [], []), [Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('Antarctica')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('-89.2')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('N/A')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('19.8')])])]), Row(('', [], []), [Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('Earth')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('-89.2')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('14')])]), Cell(('', [], []), AlignDefault(), RowSpan(1), ColSpan(1), [Plain([Str('56.7')])])])])], TableFoot(('', [], []), []))])
-
-<!-- prevent container tabs merge -->
-
 Alignments can be specified as with pipe tables, by putting colons at
 the boundaries of the separator line after the header:
 
@@ -2617,44 +2543,6 @@ or
 
 <!-- prevent container tabs merge -->
 
-Internal links are currently supported for HTML formats (including HTML
-slide shows and EPUB), LaTeX, and ConTeXt.
-
-## Images
-
-A link immediately preceded by a `!` will be treated as an image. The
-link text will be used as the image's alt text:
-
-=== "Markdown"
-
-        ![la lune](lalune.jpg "Voyage to the moon")
-        
-        ![movie reel]
-        
-        [movie reel]: movie.gif
-
-=== "Python"
-
-        Pandoc(Meta({}), [Figure(('', [], []), Caption(None, [Plain([Str('la'), Space(), Str('lune')])]), [Plain([Image(('', [], []), [Str('la'), Space(), Str('lune')], ('lalune.jpg', 'Voyage to the moon'))])]), Figure(('', [], []), Caption(None, [Plain([Str('movie'), Space(), Str('reel')])]), [Plain([Image(('', [], []), [Str('movie'), Space(), Str('reel')], ('movie.gif', ''))])])])
-
-<!-- prevent container tabs merge -->
-
-#### Extension: `implicit_figures`
-
-An image with nonempty alt text, occurring by itself in a paragraph,
-will be rendered as a figure with a caption. The image's alt text will
-be used as the caption.
-
-=== "Markdown"
-
-        ![This is the caption](/url/of/image.png)
-
-=== "Python"
-
-        Pandoc(Meta({}), [Figure(('', [], []), Caption(None, [Plain([Str('This'), Space(), Str('is'), Space(), Str('the'), Space(), Str('caption')])]), [Plain([Image(('', [], []), [Str('This'), Space(), Str('is'), Space(), Str('the'), Space(), Str('caption')], ('/url/of/image.png', ''))])])])
-
-<!-- prevent container tabs merge -->
-
 How this is rendered depends on the output format. Some output formats
 (e.g. RTF) do not yet support figures. In those formats, you'll just get
 an image in a paragraph by itself, with no caption.
-- 
2.45.2

